src/Polly.Extensions/README.md

```diff
@@ -25,9 +21,118 @@ services.AddResilienceStrategy(
     });
 
-// Use your strategy
+// Utilize the strategy
 var serviceProvider = services.BuildServiceProvider();
 var strategyProvider = serviceProvider.GetRequiredService<ResilienceStrategyProvider<string>>();
 var resilienceStrategy = strategyProvider.Get("my-key");
 ```
 
+## Telemetry Features
+
+Upon invoking the `ConfigureTelemetry` extension method, Polly begins to emit logs and metrics. Here's an example:
+
+``` csharp
+var telemetryOptions = new TelemetryOptions();
+
+// Configure logging
+telemetryOptions.LoggerFactory = LoggerFactory.Create(builder => builder.AddConsole());
+
+// Configure enrichers
+telemetryOptions.Enrichers.Add(context =>
+{
+    context.Tags.Add(new("my-custom-tag", "custom-value"));
+});
+
+// Manually handle the event
+telemetryOptions.OnTelemetryEvent = args =>
+{
+    Console.WriteLine($"Telemetry event occurred: {args.Event.EventName}");
+});
+
+var builder = new ResilienceStrategyBuilder()
+    .AddTimeout(TimeSpan.FromSeconds(1))
+    .ConfigureTelemetry(telemetryOptions) // This method enables telemetry in the builder
+    .Build();
+```
+
+Alternatively, you can use the `AddResilienceStrategy` extension which automatically adds telemetry:
+
+``` csharp
+var serviceCollection = new ServiceCollection()
+    .AddLogging(builder => builder.AddConsole())
+    .AddResilienceStrategy("my-strategy", builder => builder.AddTimeout(TimeSpan.FromSeconds(1)))
+    // Configure the default settings for TelemetryOptions
+    .Configure<TelemetryOptions>(options =>
+    {
+        // Configure enrichers
+        options.Enrichers.Add(context => context.Tags.Add(new("my-custom-tag", "custom-value")));
+
+        // Manually handle the event
+        options.OnTelemetryEvent = args =>
+        {
+            Console.WriteLine($"Telemetry event occurred: {args.Event.EventName}");
+        };
+    });
+```
+
+### Emitted Metrics
+
+The emitted metrics are emitted under the `Polly` meter name. The subsequent sections provide insights into the metrics produced by Polly. Please note that any custom enriched dimensions are not depicted in the following tables. 
+
+#### resilience-events
+
+- Type: *Counter*
+- Description: Emitted upon the occurrence of a resilience event.
+
+Dimensions:
+
+|Name|Description|
+|---| ---|
+|event-name| The name of the emitted event.| 
+|event-severity| The severity of the event (`Debug`, `Information`, `Warning`, `Error`, `Critical`).|
+|builder-name| The name of the builder corresponding to the resilience strategy.|
+|builder-instance| The instance name of the builder corresponding to the resilience strategy.|
+|strategy-name| The name of the strategy generating this event.|
+|strategy-type| The type of the strategy generating this event.|
+|operation-key| The operation key associated with the call site. |
+|result-type| The result type (`String`, `HttpResponseMessage`). |
+|exception-name| The full name of the exception assigned to the execution result (`System.InvalidOperationException`). |
```

Comment:
I wonder if we should wrap these names in these tables in backticks?

Reply:
Good idea, done.